types/prefs/item.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package prefs
+
+import (
+	"fmt"
+
+	jsonv2 "github.com/go-json-experiment/json"
+	"github.com/go-json-experiment/json/jsontext"
+	"tailscale.com/types/opt"
+	"tailscale.com/types/ptr"
+	"tailscale.com/types/views"
+	"tailscale.com/util/must"
+)
+
+// Item is a single preference item that can be configured.
+// T must either be an immutable type or implement the [views.ViewCloner] interface.
+type Item[T any] struct {
+	preference[T]
+}
+
+// ItemOf returns an [Item] configured with the specified value and [Options].
+func ItemOf[T any](v T, opts ...Options) Item[T] {
+	if cloner, ok := any(v).(views.Cloner[T]); ok {
+		v = cloner.Clone()
+	}
+	return Item[T]{preferenceOf(opt.ValueOf(v), opts...)}
+}
+
+// ItemWithOpts returns an unconfigured [Item] with the specified [Options].
+func ItemWithOpts[T any](opts ...Options) Item[T] {
```

Comment:
nit: I'd name this `UnconfiguredItem` (`WithOpts` seems redundant, since `ItemOf` doesn't have that suffix)

Reply:
`ItemWithOpts` is less characters.
I'd also prefer the related methods to share the same prefixes (e.g., `Item`), and the `WithOpts` suffix better reflects the primary use case: creating an item with the specified options set.
And unconfigured item is `prefs.Item[T]{}`.